cmd/snap-bootstrap/cmd_initramfs_mounts_cvm.go

```diff
@@ -60,42 +227,76 @@ func generateMountsModeRunCVM(mst *initramfsMountsState) error {
 		return err
 	}
 
-	// Mount rootfs
-	if err := secbootProvisionForCVM(boot.InitramfsUbuntuSeedDir); err != nil {
-		return err
-	}
-	runModeCVMKey := filepath.Join(boot.InitramfsSeedEncryptionKeyDir, "cloudimg-rootfs.sealed-key")
-	opts := &secboot.UnlockVolumeUsingSealedKeyOptions{
-		AllowRecoveryKey: true,
-	}
-	unlockRes, err := secbootUnlockVolumeUsingSealedKeyIfEncrypted(disk, "cloudimg-rootfs", runModeCVMKey, opts)
+	var partitionMounts []partitionMount
+
+	// try searching for a manifest that contains mount information
+	imageManifestFile, err := os.ReadFile(filepath.Join(boot.InitramfsUbuntuSeedDir, "EFI/ubuntu", "manifest.json"))
 	if err != nil {
-		return err
-	}
-	fsckSystemdOpts := &systemdMountOptions{
-		NeedsFsck: true,
-		Ephemeral: true,
-	}
-	if err := doSystemdMount(unlockRes.FsDevice, boot.InitramfsDataDir, fsckSystemdOpts); err != nil {
-		return err
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+
+		// XXX: if a manifest file is not found fall-back to CVM v1 behaviour
+		partitionMounts = []partitionMount{
+			{
+				Where:    boot.InitramfsDataDir,
+				GptLabel: "cloudimg-rootfs",
+				Opts: &systemdMountOptions{
+					NeedsFsck: true,
+				},
+			},
+		}
 	}
 
-	// Verify that cloudimg-rootfs comes from where we expect it to
-	diskOpts := &disks.Options{}
-	if unlockRes.IsEncrypted {
-		// then we need to specify that the data mountpoint is
-		// expected to be a decrypted device
-		diskOpts.IsDecryptedDevice = true
+	if imageManifestFile != nil {
```

Comment:
shouldn't this be an "else" of the previous "if"?

Reply:
I think it is the same logically (could `os.ReadFile` return both an error and non nil imageManifestFile?). I changed it and put it in the else block.